### PR TITLE
chore: use translate to center the resize handle

### LINKED ISSUE #35232: [Bug]: Space distribution handle appears on the zone boundary
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Space distribution handle appears on the zone boundary instead of in between 2 zones

image

### Steps To Reproduce

1. Go to [this app](https://internal.appsmith.com/app/raven-on-anvil/flow-66a392d239626131b6bec6ba/edit/widgets/40au78v0ei)
2. Try to move distribution handles
3. Observe the handles appearing on the zone boundary

### Public Sample App

_No response_

### Environment

Production

### Severity

High (Blocker to building or releasing)

### Issue video log

_No response_

### Version

cloud

### PR DESCRIPTION
**Pull request by jsartisan**
Fixes #35232

Because of how sub-pixels are handled in browsers, the resize handle was previously looking more inclined towards right. By using translate instead adjusting the width in left property, the issue seems to be resolved.

Before:
CleanShot 2024-08-13 at 17 08 47@2x

After:
CleanShot 2024-08-13 at 17 08 00@2x

</details>

## Summary by CodeRabbit

- **New Features**
	- Enhanced positioning logic for the space distribution handle, simplifying the alignment process and potentially improving rendering behavior. 

- **Documentation**
	- Added comments to clarify the rationale behind adjustments in the positioning logic and highlight the benefits of using `translateX` for centering. 

- **Bug Fixes**
	- Resolved potential pixel snapping issues related to the handle's positioning.

> [!WARNING]
> Tests have not run on the HEAD 801900efe8c65439e1b82db3fb309dcd59e461a3 yet
> <hr>Tue, 13 Aug 2024 12:05:21 UTC

### CODE REVIEW

File: app/client/src/layoutSystems/anvil/sectionSpaceDistributor/utils/spaceDistributionEditorUtils.ts [hunk 1]
```diff
@@ -56,6 +56,7 @@ const getElementsBoundingBoxValue = (ele: HTMLElement) => {
 export const getAnvilZoneBoundaryOffset = (zoneId: string) => {
   const zoneDom = document.getElementById(getAnvilWidgetDOMId(zoneId));
   if (zoneDom) {
+    // (TODO: Anvil): Why is 2 added here?
```
Comment (KelvinOm) [inline app/client/src/layoutSystems/anvil/sectionSpaceDistributor/utils/spaceDistributionEditorUtils.ts:59]: This comment is useless. Who knows?

**Review by KelvinOm — APPROVED** [review]
Could you please make a video about what changed?

File: app/client/src/layoutSystems/anvil/sectionSpaceDistributor/utils/spaceDistributionEditorUtils.ts (hunk 1 above)
Comment (jsartisan) [reply app/client/src/layoutSystems/anvil/sectionSpaceDistributor/utils/spaceDistributionEditorUtils.ts:59]: No idea but wanted to keep a track on this.

**Comment by jsartisan** [comment]
@KelvinOm added before/after images.

**Review by KelvinOm — APPROVED** [review]

**Pull request merged**